doc/developers/js_app.rst

```diff
@@ -0,0 +1,167 @@
+JavaScript Application
+======================
+
+CCF includes a native ``js_generic`` application which can execute JavaScript applications proposed through governance.
+
+CCF's JavaScript environment is similar to that in browsers, with the following differences:
+
+- only `JavaScript modules <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules>`_ are supported,
+- no `Web APIs <https://developer.mozilla.org/en-US/docs/Web/API>`_ are available,
+- module imports cannot be URLs and must be relative paths.
+
+.. _js-app-bundle:
+
+App bundle
+----------
+
+The most basic CCF JavaScript app has the following folder structure:
+
+.. code-block:: bash
+
+    $ tree --dirsfirst my-app
+    my-app
+    ├── src
+    │   └── app.js
+    └── app.json
+
+We also call this a `JavaScript app bundle` since it is the format used for deployment.
```

Comment:
I think we should try to avoid "we" and "I" in this doc, but maybe it's just me ;)

Reply:
While I agree, we also do not have documentation guidelines (yet!) or consensus on what they ought to be.